Fusion FDF decouple

clacroix12 opened the pull request:
Decouple Fusion and FDF deployments from OCP deployment. New entry-points to perform Fusion and FDF deployments on top of OCP clusters.

Issue: https://url.corp.redhat.com/12bac9c

dahorak reviewed (dismissed):
Overall LGTM, just one suggestion about the `--conf` parameter (I would propose to make it backward compatible with the other tools in ocs-ci repository and allow also `--ocsci-conf` variant).

clacroix12 commented on ocs_ci/utility/framework/fusion_fdf_init.py:
```diff
@@ -0,0 +1,231 @@
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--cluster-name", help="Name of the OCP cluster")
+        parser.add_argument("--cluster-path", help="OCP cluster directory")
+        parser.add_argument(
+            "--conf",
+            action="append",
+            default=[],
+            help="Path to config file. Repeatable.",
+        )
```
If I'm remembering correctly the inclusion of `ocsci` in the param name was because we originally also had the concept of `--cluster-conf`. However, since everything is done through `--ocsci-conf` it doesn't really make sense to differentiate.

If the team feels strongly enough about sticking with `--ocsci-conf` to maintain consistency with other entry-points then I can update it. However, I personally find the inclusion of `ocsci` in the param name to be redundant and wanted to take this opportunity to go with terminology that is as clear as possible and makes sense for these specific use cases without considering what the other entry points are doing.

ebenahar replied:
We plan for further decoupling of functionalities that are either not ODF oriented or not test oriented, such as deployment and cleanup. 
Under this assumption, I tend to agree that `--conf` makes more sense here and that this would be the first step toward those term changes (such as that `--conf` or `--cleanup-conf` would make more sense in vsphere-cleanup).

dahorak replied:
Ok, good point, so it will make sense to go the other way and change the usage of `--ocsci-conf` to `--conf` in the future.

petr-balogh commented on ocs_ci/utility/framework/fusion_fdf_init.py:
```diff
@@ -0,0 +1,237 @@
+        raise ClusterNotAccessibleError(
+            "The kubeconfig file %s doesn't exist!", kubeconfig_path
+        )
+    os.environ["KUBECONFIG"] = kubeconfig_path
```
If we can avoid the KUBECONFIG in env variable the better, as it's a main problem in running multiple threads in multicluster scenario as ENV is used across whole framework

clacroix12 replied:
I've changed this function to just check cluster access with the provided kubeconfig, without setting the env var as well as updated all `oc` commands to pass the kubeconfig directly.

clacroix12 commented on ocs_ci/framework/fusion/main.py:
```diff
@@ -0,0 +1,25 @@
+    fusion_deployment()
+
+
+@create_junit_report("FusionDeployment", "fusion_deployment")
```
I've added the ability to provide custom property key value pairs to the decorator for both the test case and the test suite. What is left is to decide which attributes we wish to add for these deployment types.

clacroix12 commented:
Rebased on master and added commits to provide custom property values to test suites. Also did some cleanup of the configuration files for both deployment types as a lot of the content in them was irrelevant with these new entry points.

Verified the latest changes locally:

Fusion Deployment Report:
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
        <testsuite name="FusionDeployment" tests="1" errors="0" failures="0" skipped="0" time="0">
                <properties>
                        <property name="run_id" value="1742946090"/>
                        <property name="cluster_path" value="/home/clacroix/clusters/cl-fusion19"/>
                        <property name="logs_dir" value="/tmp"/>
                        <property name="ocp_version" value="4.18"/>
                        <property name="fusion_version" value="v2.9.0"/>
                </properties>
                <testcase name="fusion_deployment">
                        <properties>
                                <property name="squad" value="Purple"/>
                        </properties>
                </testcase>
        </testsuite>
</testsuites>
```

FDF Deployment Report:
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
        <testsuite name="FusionDataFoundationDeployment" tests="1" errors="0" failures="0" skipped="0" time="0">
                <properties>
                        <property name="run_id" value="1742946384"/>
                        <property name="cluster_path" value="/home/clacroix/clusters/cl-fusion19"/>
                        <property name="logs_dir" value="/tmp"/>
                        <property name="ocp_version" value="4.18"/>
                        <property name="fusion_version" value="v2.9.0"/>
                        <property name="fdf_version" value="4.18.1"/>
                </properties>
                <testcase name="fusion_data_foundation_deployment">
                        <properties>
                                <property name="squad" value="Purple"/>
                        </properties>
                </testcase>
        </testsuite>
</testsuites>
```

petr-balogh commented on ocs_ci/deployment/fusion_data_foundation.py:
```diff
@@ -119,14 +121,32 @@ def setup_fdf_pre_release_deployment(self):
+        """
+        logger.info("Retrieving installed FDF version")
+        results = run_cmd(
+            f"oc describe FusionServiceInstance {constants.FDF_SERVICE_NAME} "
```
can you try with `oc get --output yaml` if it's returning also same data?

clacroix12 replied:
Fixed

clacroix12 commented:
Latest update addresses Petr's comment about using get vs describe for installed versions. It also relocated the call to create SpectrumFusion to the Fusion deployment class as opposed to the Data Foundation deployment class. This more closely mirrors the UI experience when deploying Fusion/FDF.

Reports generated when testing with pre-release configurations, so you can also see the additional properties populated for those types.

Fusion
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
        <testsuite name="FusionDeployment" tests="1" errors="0" failures="0" skipped="0" time="0">
                <properties>
                        <property name="run_id" value="1743112565"/>
                        <property name="cluster_path" value="/home/clacroix/clusters/cl-fusion21"/>
                        <property name="logs_dir" value="/tmp"/>
                        <property name="ocp_version" value="4.18"/>
                        <property name="fusion_pre_release_image" value="2.10.0-31566152"/>
                        <property name="fusion_version" value="v2.10.0-31566152"/>
                </properties>
                <testcase name="fusion_deployment">
                        <properties>
                                <property name="squad" value="Purple"/>
                        </properties>
                </testcase>
        </testsuite>
</testsuites>
```

FDF:
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
        <testsuite name="FusionDataFoundationDeployment" tests="1" errors="0" failures="0" skipped="0" time="0">
                <properties>
                        <property name="run_id" value="1743113072"/>
                        <property name="cluster_path" value="/home/clacroix/clusters/cl-fusion21"/>
                        <property name="logs_dir" value="/tmp"/>
                        <property name="ocp_version" value="4.18"/>
                        <property name="fdf_image_tag" value="v4.18"/>
                        <property name="fdf_pre_release_registry" value="docker-na-public.artifactory.swg-devops.com/hyc-abell-devops-team-dev-docker-local/df"/>
                        <property name="fdf_pre_release_image_digest"/>
                        <property name="fusion_version" value="v2.10.0-31566152"/>
                        <property name="fdf_version" value="4.18.1"/>
                </properties>
                <testcase name="fusion_data_foundation_deployment">
                        <properties>
                                <property name="squad" value="Purple"/>
                        </properties>
                </testcase>
        </testsuite>
</testsuites>
```

clacroix12 commented:
Noticed `fdf_pre_release_image_digest` had no value because this is set during the deployment and not before. Updated the logic to grab this value after deployment as well. Confirmed in the report xml that the value is now being populated:

```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
        <testsuite name="FusionDataFoundationDeployment" tests="1" errors="0" failures="0" skipped="0" time="0">
                <properties>
                        <property name="run_id" value="1743118726"/>
                        <property name="cluster_path" value="/home/clacroix/clusters/cl-fusion22"/>
                        <property name="logs_dir" value="/tmp"/>
                        <property name="ocp_version" value="4.18"/>
                        <property name="fdf_image_tag" value="v4.18"/>
                        <property name="fdf_pre_release_registry" value="docker-na-public.artifactory.swg-devops.com/hyc-abell-devops-team-dev-docker-local/df"/>
                        <property name="fusion_version" value="v2.9.0"/>
                        <property name="fdf_version" value="4.18.1"/>
                        <property name="fdf_pre_release_image_digest" value="sha256:4ad07121a20bba3c68366b91d0f2164130b45c96866f2cadda57b201cc4a510a"/>
                </properties>
                <testcase name="fusion_data_foundation_deployment">
                        <properties>
                                <property name="squad" value="Purple"/>
                        </properties>
                </testcase>
        </testsuite>
</testsuites>
```

clacroix12 commented:
/cherry-pick release-4.18
